content/en/docs/resources/guides/extending-jx/introduction.md

```diff
@@ -56,6 +56,35 @@ Planned features include:
 * `jx get apps` - the ability to list all apps that can be installed
 * integration for bash completion
 
+## Structure of an App chart
+
+In addition to the structure of a regular helm chart a Jenkins X App chart contains the following:
+* `values.schema.yaml`: a JSON schema extended with conditions, field values and questions that is used to point out and prompt the user for customizable values the app should be installed with.
+* `templates/app.yaml`: a Jenkins X `App` custom resource: allows to define a `values.schema.yaml` transformation which can be used to inject cluster-specific values as for instance its domain and TLS settings (which can be obtained by mounting the ConfigMap `ingress-config`).
+
+When `jx add app` is called the corresponding helm chart is fetched, its `values.schema.yaml` transformed using the `App` resource and the user prompted for the contained questions (non-const fields).
```

Comment:
I was wondering whether it makes sense to tie in more the generated CLI reference and link commands like this to their reference. There you can see options, examples, etc

Reply:
The beginning of this document would need updating as well. It talks about Jenkinsfile (aka static master). Also there is a redirect/alias from addons which makes me believe that this was originally the addons docs and got partly converted to the Apps docs. Personally I would remove everything up to the first heading "Install and Configuration Framework". WDYT?

Reply:
yes, i've also noticed this page is quite outdated in unorganized. however I didn't want to remove stuff from it.